partner_multi_relation/models/res_partner.py

```diff
@@ -184,11 +184,14 @@ def action_view_relations(self):
                 'partner_multi_relation.action_res_partner_relation_all'
             ).read()[0]
             action['domain'] = [('id', 'in', relation_ids.ids)]
-            action['context'].\
-                update({'search_default_this_partner_id': contact.id,
-                        'default_this_partner_id': contact.id,
+            context = action.get('context', '{}').strip()[1:-1]
+            elements = context.split(',') if context else []
+            to_add = ["""'search_default_this_partner_id': {0},
+                        'default_this_partner_id': {0},
                         'active_model': 'res.partner',
-                        'active_id': contact.id,
-                        'active_ids': [contact.id],
-                        'active_test': False})
+                        'active_id': {0},
+                        'active_ids': [{0}],
+                        'active_test': False""".format(contact.id)]
+            context = '{' + ', '.join(elements + to_add) + '}'
```

Comment:
Unfortunately this is still not foolproof. The dictionary update would replace possibly existing values for the keywords added. You have to clear the elements array from any element that has a key, that is also in the to_add  array. As the button is not working at all at the moment I let it pass, but note that this might need to be repaired later on.